wavefront/resource_ingestion_policy.go

```diff
@@ -75,48 +181,88 @@ func resourceIngestionPolicyRead(
 		return fmt.Errorf("error finding ingestion policy, %s. %s", d.Id(), err)
```

Comment:
Hmm, looks like that is pre-me as well. 🤔 In retrospect it is kind of funny, the conditional right before this error handler is handling the NotFound explicitly and it returns immediately.

I guess it technically the client has failed to "get" the policy for you but for networking reasons, not because the returned payload said it couldn't find it. That's probably what the original wording was going for, just not super well thought out.

Reply:
I am trying to think of a new error message. It seems like that method goes fairly deep and maybe could fail in non network ways? I wonder if "network error" is too specific and sometimes err might contain a different kind of error, leading to a misleading error message. Could it fail generating a request payload for example (pretty simple request in this specific case but, just thinking broadly here.)?